app/src/main/java/io/homeassistant/companion/android/widgets/SingleItemArrayAdapter.kt

```diff
@@ -4,12 +4,81 @@ import android.content.Context
 import android.view.View
 import android.view.ViewGroup
 import android.widget.ArrayAdapter
+import android.widget.Filter
 import android.widget.TextView
+import java.util.Comparator
 
 class SingleItemArrayAdapter<T>(
     context: Context,
     private val createText: (T?) -> String
 ) : ArrayAdapter<T>(context, android.R.layout.simple_list_item_1) {
+
+    private var filterItems = ArrayList<T>()
+
+    private var containsFilter: Filter = object : Filter() {
+        override fun performFiltering(constraint: CharSequence?): FilterResults {
+            val result = FilterResults()
+
+            constraint?.let {
+                val containSeq = ArrayList<T>()
+
+                for (i in 0 until filterItems.size) {
+                    val item = filterItems[i]
+                    if (createText(item).contains(constraint)) {
+                        containSeq.add(item!!)
+                    }
+                }
+                result.values = containSeq
+                result.count = containSeq.size
+            }
+            return result
+        }
+
+        override fun publishResults(constraint: CharSequence?, results: FilterResults?) {
+            results?.let {
+                if (results.count > 0) {
```

Comment:
Interestingly, that does not seem to be the case.  However, to avoid that bug if it's just behaving oddly for me, I will change it so `clear()` and `notifyDataSetChanged()` happen whether or not `results.count > 0` is true or not.

Reply:
Changed in commit b3d9d06